src/components/Cart/components/CartOrderSummary/components/CartPricingSummary/components/CartTotalPriceAndDelivery/CartTotalPriceAndDelivery.tsx

```diff
@@ -0,0 +1,32 @@
+import { FC } from "react";
+import { getCartDeliveryTime, getPriceWithPromo, getPriceWithTax, getSubtotalPrice } from "@/utils";
+import { CartPayload, CartProduct, CartPromocode } from "@/common/types";
+import styles from "./CartTotalPriceAndDelivery.module.scss";
+
+type Props = {
+  cartProducts: CartProduct[];
+  cartProductsPayload: CartPayload[];
+  orderPromo: CartPromocode;
+};
+
+export const CartTotalPriceAndDelivery: FC<Props> = ({ cartProductsPayload, cartProducts, orderPromo }) => {
+  const getTotalPrice = () => {
+    const subtotalPrice = getSubtotalPrice(cartProductsPayload);
+    const taxIncludedPrice = getPriceWithTax(subtotalPrice);
+    const priceWithPromo = getPriceWithPromo(subtotalPrice, orderPromo.discount);
+
+    return priceWithPromo ? subtotalPrice + taxIncludedPrice - priceWithPromo : subtotalPrice + taxIncludedPrice;
```

Comment:
It is better to split into two variables subtotalPrice + taxIncludedPrice - priceWithPromo and subtotalPrice + taxIncludedPrice